### PR TITLE
First pass on filtering of accessors

### PR DESCRIPTION
**Pull request by marcingrzejszczak**

### CODE REVIEW

File: context-propagation-api/src/main/java/io/micrometer/contextpropagation/Namespace.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        return getClass().getName();
+    }
+
+    /**
+     * Creates a new instance of a {@link Namespace}.
+     *
+     * @param key key under which this namespace is to be registered
+     * @return new namespace
+     */
+    static Namespace create(String key) {
```
Comment (jonatan-ivanov) [inline context-propagation-api/src/main/java/io/micrometer/contextpropagation/Namespace.java:52]: What do you think about `Namespace.of`?

File: context-propagation-api/src/test/java/io/micrometer/contextpropagation/ObservationThreadLocalAccessor.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        }
+    }
+
+    @Override
+    public void restoreValues(ContextContainer container) {
+        if (namespaceAccessor.isPresent(container)) {
+            ObservationStore store = namespaceAccessor.getStore(container);
+            String observation = store.getObservation();
+            ObservationThreadLocalHolder.holder.set(observation);
+            System.out.println("Observation [" + observation + "]");
```
Comment (jonatan-ivanov) [inline context-propagation-api/src/test/java/io/micrometer/contextpropagation/ObservationThreadLocalAccessor.java:40]: Remove this?

File: context-propagation-api/src/test/java/io/micrometer/contextpropagation/ObservationThreadLocalAccessor.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        }
+    }
+
+    @Override
+    public void resetValues(ContextContainer container) {
+        namespaceAccessor.getRequiredStore(container).close();
+    }
```
Comment (ttddyy) [inline context-propagation-api/src/test/java/io/micrometer/contextpropagation/ObservationThreadLocalAccessor.java:47]: The above `restoreValues()` method checks namespace existence by `namespaceAccessor.isPresent`, but here uses `getRequiredStore()`.
It seems these methods are called from `restoreThreadLocalValues()` and closing the returned `Scope`.

The pre-requisite for `restoreThreadLocalValues()` to properly work is that `captureValues()` needs to create a `Store` in the `namespaceAccessor`.

Per implementation in `ObservationThreadLocalAccessor#captureValues`, `ObservationStore` is only created when `ObservationThreadLocalHolder.holder.get()` is not null.

In the unit test, if I comment out `ObservationThreadLocalHolder.holder.set("hello");`, it fails with followings:
```java
Store must be set
java.lang.IllegalStateException: Store must be set
	at io.micrometer.contextpropagation.NamespaceAccessor.getRequiredStore(NamespaceAccessor.java:78)
	at io.micrometer.contextpropagation.ObservationThreadLocalAccessor.resetValues(ObservationThreadLocalAccessor.java:46)
	at io.micrometer.contextpropagation.SimpleContextContainer.lambda$restoreThreadLocalValues$6(SimpleContextContainer.java:105)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
	at io.micrometer.contextpropagation.SimpleContextContainer.lambda$restoreThreadLocalValues$7(SimpleContextContainer.java:105)
	at io.micrometer.contextpropagation.SimpleThreadLocalAccessorTests.should_capture_thread_local_values(SimpleThreadLocalAccessorTests.java:38)
	...
```

So, I think here also should be guarded like the above method.

```java
    @Override
    public void resetValues(ContextContainer container) {
		if (namespaceAccessor.isPresent(container)) {
			namespaceAccessor.getRequiredStore(container).close();
		}
    }
```

File: context-propagation-api/src/main/java/io/micrometer/contextpropagation/Namespace.java (hunk 1 above)
Comment (marcingrzejszczak) [reply context-propagation-api/src/main/java/io/micrometer/contextpropagation/Namespace.java:52]: I was consistent with other factory methods that we have that are called `create`

File: context-propagation-api/src/test/java/io/micrometer/contextpropagation/ObservationThreadLocalAccessor.java (hunk 3 above)
Comment (marcingrzejszczak) [reply context-propagation-api/src/test/java/io/micrometer/contextpropagation/ObservationThreadLocalAccessor.java:47]: Great catch!

**Pull request merged**